aspnet/fundamentals/dist-cache/sample/src/DistCacheSample/Startup.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Http;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Caching.Redis;
+using Microsoft.Extensions.Caching.SqlServer;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.OptionsModel;
+using System;
+using System.Text;
+
+namespace DistCacheSample
+{
+    public class Startup
+    {
+        /// <summary>
+        /// Use LocalCache (Memory) in Development
+        /// </summary>
+        /// <param name="services"></param>
+        public void ConfigureDevelopmentServices(IServiceCollection services)
+        {
+            services.AddSingleton<IMemoryCache>(serviceProvider =>  new MemoryCache(new MemoryCacheOptions()));
+            services.AddSingleton<IDistributedCache, LocalCache>();
+        }
+
+        /// <summary>
+        /// Use Redis Cache in Staging
+        /// </summary>
+        /// <param name="services"></param>
+        public void ConfigureStagingServices(IServiceCollection services)
+        {
+            // use Redis
+            services.AddSingleton<IDistributedCache>(serviceProvider => 
+                new RedisCache(new RedisCacheOptions
+                {
+                    Configuration = "localhost",
+                    InstanceName = "SampleInstance"
+                }));
+        }
+
+        /// <summary>
+        /// Use SQL Server Cache in Production
+        /// </summary>
+        /// <param name="services"></param>
+        public void ConfigureProductionServices(IServiceCollection services)
+        {
+            // Use SQL Server
+            services.AddSingleton<IDistributedCache>(serviceProvider =>
+            new SqlServerCache(new CacheOptions(new SqlServerCacheOptions()
+            {
+                ConnectionString = @"Data Source=(localdb)\v11.0;Initial Catalog=DistCache;Integrated Security=True;",
+                SchemaName = "dbo",
+                TableName = "TestCache"
+            })));
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseIISPlatformHandler();
+
+            var serverStartTimeString = DateTime.Now.ToString();
+            byte[] val = Encoding.UTF8.GetBytes(serverStartTimeString);
+
+            var cache = app.ApplicationServices.GetService(typeof(IDistributedCache)) as IDistributedCache;
+            cache.Set("serverStartTime", val);
```

Comment:
This seems like a very strange piece of data to put in a distributed cache. Every server would reset it, and it would only be accurate for the last server started. At least rename it to lastServerStartTime. You could also change it to a counter for the number of server starts.